vortex-expr/src/context.rs

```diff
@@ -0,0 +1,148 @@
+use std::any::Any;
+use std::iter;
+use std::sync::Arc;
+
+use itertools::Itertools;
+use vortex_array::aliases::hash_map::HashMap;
+use vortex_array::{Array, ArrayRef};
+use vortex_dtype::DType;
+use vortex_error::{VortexError, VortexExpect, VortexResult, vortex_err};
+
+use crate::IDENTITY_IDENTIFIER;
+
+pub type Identifier = Arc<str>;
+pub type ExprScope<T> = HashMap<Identifier, T>;
+
+#[derive(Clone, Default)]
+pub struct ValuesScope(ExprScope<ArrayRef>);
```

Comment:
I wouldn't bother wrapping / exposing these types, keep them internal to Context (Scope)